src/mmap_view_sync.rs

```diff
@@ -13,7 +14,7 @@ use std::sync::Arc;
 /// The view may be split into disjoint ranges, each of which will share the
 /// underlying memory map.
 pub struct MmapViewSync {
-    inner: Arc<UnsafeCell<MmapMut>>,
+    inner: Arc<Mutex<UnsafeCell<MmapMut>>>,
```

Comment:
I wonder if we could actually get rid of the `UnsafeCell` completely now that a `Mutex` protects from concurrent access?

Reply:
I tried to remove `UnsafeCell` completely, but in `fn inner(&self) -> &MmapMut` and `fn inner_mut(&self) -> &mut MmapMut` the compiler says `cannot return value referencing temporary value`. It seems we still need this `UnsafeCell`.